loop/storage/redis.js

```diff
@@ -796,16 +804,23 @@ RedisStorage.prototype = {
    * @param {Function} A callback that will be called when the action
    *                   has been processed.
    **/
-  touchHawkSession: function(hawkIdHmac, callback) {
+  touchHawkSession: function(userMac, hawkIdHmac, callback) {
+    if (isUndefined(userMac, "hawkIdHmac", callback)) return;
     if (isUndefined(hawkIdHmac, "hawkIdHmac", callback)) return;
 
     var self = this;
     var multi = self._client.multi();
-    var hawkSessionDuration = self._settings.hawkSessionDuration;
+    var ttl = self._settings.hawkSessionDuration;
+
+    multi.expire('userid.' + hawkIdHmac, ttl);
```

Comment:
I find the multi.expire more explicit. to have a multi per key rather than a list and a forEach. Don't you?

`oauth.token` has to stand only for the login process not during the whole livetime of the user account.